lib/gtfsplus/reducers/gtfsplus.js

```diff
@@ -53,49 +60,76 @@ const gtfsplus = (state = {
       return update(state, {
         feedVersionId: {$set: action.feedVersionId},
         timestamp: {$set: action.timestamp},
-        tableData: {$set: newTableData}
+        tableData: {$set: newTableData},
+        pageCount: {$set: Math.ceil(newTableData.length / state.recordsPerPage)}
       })
     case 'ADD_GTFSPLUS_ROW':
+      const newRow = constructNewGtfsPlusRow(action.payload.tableId)
+      const tableExists = action.payload.tableId in state.tableData
+      // add rowIndex for use with tracking row for deletion/updating values
+      newRow.rowIndex = tableExists
+        ? state.tableData[state.activeTableId].length
+        : 1
       // create this table if it doesn't already exist
-      if (!(action.tableId in state.tableData)) {
+      if (!tableExists) {
         return update(state,
           {tableData:
-            {$merge: {[action.tableId]: [action.rowData]}}
+            {$merge: {[action.payload.tableId]: [newRow]}}
           }
         )
       }
+      const pageCountPostAdd = Math.ceil((state.tableData[state.activeTableId].length + 1) / state.recordsPerPage)
+      // Anytime a user adds a new row, the expected behavior would be to see that row,
+      // so we set the current page to page count if it is less than the new count
+      const currentPagePostAdd = pageCountPostAdd > state.currentPage
+        ? pageCountPostAdd
+        : state.currentPage
       // otherwise, add it to the exising table
       return update(state, {
         tableData: {
-          [action.tableId]: {
-            $push: [action.rowData]
+          [action.payload.tableId]: {
+            $push: [newRow]
           }
-        }
+        },
+        pageCount: {$set: pageCountPostAdd},
+        currentPage: {$set: currentPagePostAdd}
       })
     case 'UPDATE_GTFSPLUS_FIELD':
       return update(state, {
         tableData: {
-          [action.tableId]: {
-            [action.rowIndex]: {
-              [action.fieldName]: {
-                $set: action.newValue
+          [action.payload.tableId]: {
+            [action.payload.rowIndex]: {
+              [action.payload.fieldName]: {
```

Comment:
An error happens here after trying to set a stop_id in a new row in the stop_attributes table.

Reply:
This was due to an off-by-one issue when adding a new row to an empty table (rowIndex was incorrectly set).